litestar/contrib/pydantic/pydantic_dto_factory.py

```diff
@@ -16,21 +17,36 @@
 
     from litestar.typing import FieldDefinition
 
-
+# isort: off
 try:
-    import pydantic
-    from pydantic import ValidationError
-
-    if pydantic.VERSION.startswith("2"):
-        from pydantic_core import PydanticUndefined
-    else:  # pragma: no cover
-        from pydantic.fields import Undefined as PydanticUndefined  # type: ignore
-except ImportError as e:
+    from pydantic import v1 as pydantic_v1
+    import pydantic as pydantic_v2
+    from pydantic.v1 import ValidationError as ValidationErrorV1
+    from pydantic import ValidationError as ValidationErrorV2
+
+
+except ImportError:
+    import pydantic as pydantic_v1  # type: ignore[no-redef]
+
+    pydantic_v2 = Empty  # type: ignore[assignment]
+    from pydantic import ValidationError as ValidationErrorV1  # type: ignore[assignment]
+
+    ValidationErrorV2 = ValidationErrorV1  # type: ignore[assignment, misc]
+
+
+except ImportError as e:  # noqa: B025
     raise MissingDependencyException("pydantic") from e
+# isort: on
 
-__all__ = ("PydanticDTO",)
+if pydantic_v2 is not Empty:  # type: ignore[comparison-overlap]
```

Comment:
Mypy doesn't like this at all, but it also doesn't like "redefining" type aliases within an `if`/`else` branch so I'm not really sure how to best deal with this.